### PR TITLE
Dps Fetch Refactor #006: Add array-based data classes. Change camel case defaults

### PR DESCRIPTION
**Pull request by haakonvt**

### CODE REVIEW

File: cognite/client/data_classes/time_series.py [hunk 1]
```diff
@@ -75,43 +75,26 @@ def __init__(
         self.legacy_name = legacy_name
         self._cognite_client = cast("CogniteClient", cognite_client)
 
-    def plot(
```
Comment (erlendvollset) [inline cognite/client/data_classes/time_series.py:78]: If removing these plot methods, that should be noted in the changelog and I also believe the docs mention some stuff about these methods.

File: cognite/client/data_classes/assets.py [hunk 2]
```diff
@@ -203,7 +203,7 @@ def dump(self, camel_case: bool = False) -> Dict[str, Any]:
         return result
 
     def to_pandas(
-        self, expand: Sequence[str] = ("metadata", "aggregates"), ignore: List[str] = None, camel_case: bool = True
```
Comment (erlendvollset) [inline cognite/client/data_classes/assets.py:206]: Changing defaults here should be noted in the changelog as a breaking change.

File: cognite/client/data_classes/time_series.py (hunk 1 above)
Comment (haakonvt) [reply cognite/client/data_classes/time_series.py:78]: Yup, see #1049

File: cognite/client/data_classes/assets.py (hunk 2 above)
Comment (haakonvt) [reply cognite/client/data_classes/assets.py:206]: On it ⚡

Comment (haakonvt) [reply cognite/client/data_classes/assets.py:206]: Done, https://github.com/cognitedata/cognite-sdk-python/pull/1047/files#diff-1d2462c0f2a2dfee857bfde4966c1f353c6555def8165897cf15ff064fdea090

File: cognite/client/data_classes/time_series.py [hunk 3]
```diff
@@ -122,19 +105,21 @@ def latest(self) -> Optional["Datapoint"]:  # noqa: F821
...
         Returns:
             Datapoint: A datapoint object containing the value and timestamp of the first datapoint.
         """
         identifier = Identifier.load(self.id, self.external_id).as_dict()
-        dps = self._cognite_client.datapoints.retrieve(**identifier, start=0, end="now", limit=1)
+        dps = self._cognite_client.time_series.data.retrieve(
+            **identifier, start=MIN_TIMESTAMP_MS, end=MAX_TIMESTAMP_MS, limit=1
+        )
         if len(dps) > 0:
             return list(dps)[0]
```
Comment (haakonvt) [inline cognite/client/data_classes/time_series.py:124]: Note to self: drop list conversion

File: cognite/client/data_classes/datapoints.py [hunk 4]
```diff
@@ -60,32 +116,201 @@ def __init__(
...
         id (int): Id of the timeseries the datapoints belong to
-        external_id (str): External id of the timeseries the datapoints belong to (Only if id is not set)
+        external_id (str): External id of the timeseries the datapoints belong to
         is_string (bool): Whether the time series is string valued or not.
         is_step (bool): Whether the time series is a step series or not.
         unit (str): The physical unit of the time series.
-        timestamp (List[Union[int, float]]): The data timestamps in milliseconds since the epoch (Jan 1, 1970). Can be negative to define a date before 1970. Minimum timestamp is 1.1.1900 00:00:00 UTC
-        value (List[Union[int, str, float]]): The data values. Can be String or numeric depending on the metric
+        timestamp (List[Union[int, float]]): The data timestamps in milliseconds since the epoch (Jan 1, 1970). Can be negative to define a date before 1970. Minimum timestamp is 1900.01.01 00:00:00 UTC
+        value (List[Union[int, str, float]]): The data values. Can be string or numeric depending on the metric
```
Comment (haakonvt) [inline cognite/client/data_classes/datapoints.py:313]:
```suggestion
        value (Union[List[str], List[float]]): The data values. Can be string or numeric
```

File: cognite/client/data_classes/assets.py (hunk 2 above)
Comment (haakonvt) [reply cognite/client/data_classes/assets.py:206]: Done in #1049

**Comment by erlendvollset** [comment]
What's the rationale for changing the camel case defaults?

**Review by erlendvollset — APPROVED** [review]
LGTM

**Comment by haakonvt** [comment]
> What's the rationale for changing the camel case defaults?

@erlendvollset Out of the 14 `to_pandas` methods in the SDK; 7 accept a `camel_case` parameter - out of which 4 has the default `True` and the remaining 3 uses `False`. I think choosing one of these is a good way to make the SDK feel more unified - and went with _not camel case_ since this is Python 😜

**Pull request merged**